### DISCUSSION TITLE
What is the clean way to use Devel::Cover on multiple test files?

### DISCUSSION THREAD

**Discussion by tuxayo** [pjcj/Devel--Cover#343] [Q&A]
Not to run the whole suite, but just to run individual files and a directory

Is this a decent way to proceed? `PERL5OPT=-MDevel::Cover prove t/MyTest.t t/MyTestDir`
It works but there is no output from Devel::Cover so I miss messages like #342

I found this:
https://stackoverflow.com/questions/11212425/how-to-use-develcover-with-prove

`cover -t -make 'prove -Ilib -r t/MyTest.t t/MyTestDir ; exit $?'`
Same as the 1st attempt. Works, but silences output of Devel::Cover

`prove --exec 'perl -MDevel::Cover -Ilib' t/MyTest.t t/MyTestDir`
It works but I have duplicate warnings (e.g. of #342) for each test file. Maybe that's normal?

Here, using a makefile doesn't really fit because it's to review patches and each time I'm running the few relevant test files/directories. To avoid running the whole huge test suite. And it's really great to have a tool as powerful as Devel::Cover to easily check that the new tests are enough to cover the new code :D

So I'm trying to be sure to use the tool right and not cause unintended bad effects.

**Comment by jkeenan** [comment]
> Not to run the whole suite, but just to run individual files and a directory
> 

I don't really understand what you mean by "just ... [running] individual files and a directory".

The point of coverage analysis, as I understand it, is to have one or more test files which exercise the statements, branches, etc., of a distinct file or library.  The test files either exercise functions exported by a library or (I presume) call a given program with different command-line arguments.

It appears that you want to run *some, but not all* test files from a larger group ("suite") of test files.  I don't see where that helps you learn what statements in your program are as yet unexercised.  At best it tells you what statements have been exercised *so far*.

Can you clarify what you're trying to do?
